docs/migration_guides/opflow_migration.rst

```diff
@@ -0,0 +1,1156 @@
+=======================
+Opflow Migration Guide
+=======================
+
+*Jump to* `TL;DR`_.
+
+Background
+----------
+
+The :mod:`~qiskit.opflow` module was originally introduced as a layer between circuits and algorithms, a series of building blocks
+for quantum algorithms research and development.
+
+The recent release of the :mod:`~qiskit.primitives` introduced a new interface for interacting with backends. Now, instead of
+preparing a circuit to execute with a ``backend.run()`` type of method, the algorithms can leverage the :class:`~Sampler` and
+:class:`~Estimator` primitives, send parametrized circuits and observables, and directly receive quasi-probability distributions or
+expectation values (respectively). This workflow simplifies considerably the pre-processing and post-processing steps
+that previously relied on opflow; encouraging us to move away from opflow, and find new paths of developing algorithms based on
+the :mod:`~qiskit.primitives` interface and the :mod:`~qiskit.quantum_info` module.
+
+This guide traverses the opflow submodules and provides either a direct alternative
+(i.e. using :mod:`~qiskit.quantum_info`), or an explanation of how to replace their functionality in algorithms.
+
+TL;DR
+-----
+.. note::
+
+    The new :mod:`~qiskit.primitives`, in combination with the :mod:`~qiskit.quantum_info` module, have superseded
+    functionality of :mod:`~qiskit.opflow`. Thus, the latter is being deprecated.
+
+Index
+-----
+This guide covers the migration from these opflow sub-modules:
+
+**Operators**
+
+- `Operator Base Class`_
+- `Operator Globals`_
+- `Primitive and List Ops`_
+- `State Functions`_
+
+**Converters**
+
+- `Converters`_
+- `Evolutions`_
+- `Expectations`_
+
+**Gradients**
+
+- `Gradients`_
+
+
+Operator Base Class
+-------------------
+*Back to* `Index`_
+
+The :class:`~opflow.OperatorBase` abstract class can generally be replaced with :class:`~quantum_info.BaseOperator`, keeping in
+mind that :class:`~quantum_info.BaseOperator` is more generic than its opflow counterpart.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Opflow
+     - Alternative
+   * - :class:`~opflow.OperatorBase`
+     - :class:`~quantum_info.BaseOperator`
+
+..  tip::
+
+    Please note that:
+
+    1. :class:`~opflow.OperatorBase` implements a broader algebra mixin. Some operator overloads are not available in
+    :class:`~quantum_info.BaseOperator`.
+
+    2. :class:`~opflow.OperatorBase` also implements methods such as ``.to_matrix()`` or ``.to_spmatrix()``, which are only found
+    in some of the :class:`~quantum_info.BaseOperator` subclasses.
+
+    See API reference for more information.
+
+
+Operator Globals
```

Comment:
A general comment about headings etc. The above index/main headings - that seems to get lost when scrolling through the text as many headings seem to be text of a similar size making the structure when scrolling less than obvious. There are even examples with the same size font.
I guess it might be possible to break the text into different documents which would keep one within a given top level context too.

Reply:
I am not a big fan of breaking the text into different documents, but I have decided to change the format of the twisties, and that will hopefully also help to navigate the text better.